elementary/monitor/data_monitoring/data_monitoring_alerts.py

```diff
@@ -101,12 +167,45 @@ def _send_test_message(self):
         )
         logger.info("Sent the test message.")
 
-    def _send_alerts(self, alerts: Alerts):
-        self._send_alerts_to_slack(alerts.tests.get_all(), TestAlert.TABLE_NAME)
-        self._send_alerts_to_slack(alerts.models.get_all(), ModelAlert.TABLE_NAME)
-        self._send_alerts_to_slack(
-            alerts.source_freshnesses.get_all(), SourceFreshnessAlert.TABLE_NAME
+    def _send_alerts(self, alerts: Alerts, dont_update_as_sent=True):
+        all_alerts_to_send = alerts.get_all()
+        # TODO when pushing this to master, dont_update_as_sent should default to FALSE
+        if not all_alerts_to_send:
+            self.execution_properties["sent_alert_count"] = self.sent_alert_count
+            return
+
+        sent_alert_ids_and_tables: List[Tuple[str, str]] = []
+
+        alerts_groups: List[GroupOfAlerts] = self._group_alerts_per_config(
+            all_alerts_to_send
         )
+        alerts_with_progress_bar = alive_it(alerts_groups, title="Sending alerts")
+        for alert_group in alerts_with_progress_bar:
+            self._fix_owners_and_subscribers(alert_group)
+            alert_msg = alert_group.to_slack()
+            sent_successfully = self.slack_client.send_message(
+                channel_name=alert_group.channel_destination,
+                message=alert_msg,
+            )
+            alerts_ids_and_tables = [
+                (alert.id, alert.alerts_table) for alert in alert_group.alerts
+            ]
+            if sent_successfully:
+                sent_alert_ids_and_tables.extend(alerts_ids_and_tables)
+            else:
+                logger.error(
+                    f"Could not send the alert[s] - {[alert_id_and_table[0] for alert_id_and_table in alerts_ids_and_tables]}. Full alert: {json.dumps(dict(alert_msg))}"
+                )
+                self.success = False
+        if not dont_update_as_sent:
```

Comment:
I think it should be more of a debugging line you add once you develop instead of a valid feature in the system that could be enabled by accident 🤔

Reply:
Accept. (don't agree it but don't have any valid reason why so  🤷 )

Reply:
OK

Reply:
done